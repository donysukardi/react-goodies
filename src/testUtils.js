const propsToDataAttrs = props =>
  Object.entries(props).reduce((acc, [key, value]) => {
    acc[`data-${String(key).toLowerCase()}`] = value
    return acc
  }, {})

export {propsToDataAttrs}
