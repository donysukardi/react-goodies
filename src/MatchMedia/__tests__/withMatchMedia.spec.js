/* eslint-disable global-require */
import React from 'react'
import { cleanup, render } from 'react-testing-library'
import { propsToDataAttrs } from '../../testUtils'

const Target = props => (
  <div data-testid="targetContainer" {...propsToDataAttrs(props)} />
)

describe('withMatchMedia', () => {
  let mockJson2mq = null
  let withMatchMedia = null

  afterEach(cleanup)

  beforeEach(() => {
    mockJson2mq = jest.fn()

    jest.mock('json2mq', () => mockJson2mq)
    jest.resetModules()

    withMatchMedia = require('../withMatchMedia').default
  })

  afterAll(() => {
    jest.unmock('json2mq')
  })

  describe('`window.matchMedia` is supported', () => {
    let originMatchMedia = null

    beforeAll(() => {
      originMatchMedia = global.matchMedia
    })

    beforeEach(() => {
      global.matchMedia = () => {}
    })

    afterAll(() => {
      global.matchMedia = originMatchMedia
    })

    it('should just pass props through', () => {
      const EnhancedTarget = withMatchMedia()(Target)
      const { getByTestId } = render(<EnhancedTarget a={1} b={2} c={3} />)

      expect(getByTestId('targetContainer')).toMatchSnapshot()
    })

    describe('display name', () => {
      const origNodeEnv = process.env.NODE_ENV

      afterAll(() => {
        process.env.NODE_ENV = origNodeEnv
      })

      it('should wrap display name in non-production env', () => {
        process.env.NODE_ENV = 'test'

        const EnhancedTarget = withMatchMedia()(Target)
        render(<EnhancedTarget />)
        expect(EnhancedTarget.displayName).toMatchSnapshot()
      })

      it('should not wrap display name in production env', () => {
        process.env.NODE_ENV = 'production'

        const EnhancedTarget = withMatchMedia()(Target)
        render(<EnhancedTarget />)
        expect(EnhancedTarget.displayName).toMatchSnapshot()
      })
    })
  })

  describe('`window.matchMedia` is not supported', () => {
    afterEach(cleanup)

    it('should just pass Target component through', () => {
      const EnhancedTarget = withMatchMedia()(Target)
      const { getByTestId } = render(<EnhancedTarget a={1} b={2} c={3} />)

      expect(getByTestId('targetContainer')).toMatchSnapshot()
    })
  })
})
