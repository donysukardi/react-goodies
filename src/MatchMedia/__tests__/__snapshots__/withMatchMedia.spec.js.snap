// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withMatchMedia \`window.matchMedia\` is not supported should just pass Target component through 1`] = `
<div
  data-a="1"
  data-b="2"
  data-c="3"
  data-testid="targetContainer"
/>
`;

exports[`withMatchMedia \`window.matchMedia\` is supported display name should not wrap display name in production env 1`] = `undefined`;

exports[`withMatchMedia \`window.matchMedia\` is supported display name should wrap display name in non-production env 1`] = `"withMatchMedia(Target)"`;

exports[`withMatchMedia \`window.matchMedia\` is supported should just pass props through 1`] = `
<div
  data-a="1"
  data-b="2"
  data-c="3"
  data-testid="targetContainer"
/>
`;
